### ISSUE TITLE
Radioking tx18s suport

### ISSUE THREAD

**Issue by txemyprieto** [opentx/opentx#8091]
*Before suggesting a new feature here, please search the open issue if there is already a request for the feature*

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by pitersonfascist** [comment]
+1 vote

**Comment by brainbubblersbest** [comment]
https://www.open-tx.org/newradios.html

**Comment by d-hartmann** [comment]
Not all radios should get assigned to the latest branch... In general +1, but here, please complete the available radios and THAN assigne FR to new branch

**Comment by sfjuocekr** [comment]
I flashed it with the T18 build and it works no problem, you just won\t be able to use debug analog to view the touchscreen position/cursor.

edit:

I bothed up a patch, but I can't seem to get any builds that I make working on this thing and flashing back to the T18 obtained from the compantion works as usual.

This should apply cleanly to the master branch: http://downloads.sjoer.org/patch.diff

But, Im not sure what Im doing wrong. If I try to build for the T18 target it doesnt work either. It gets stuck on the OpenTX screen every time. Hopefully I saved someone a little time to get this thing supported.

**Comment by sfjuocekr** [comment]
So yea, instead of building it from the master 2.3 branch I downloaded the release "opentx-release-2.3.11.tar.gz" and extracted it.

I'm trying to build it like this:

cmake -DPCB=X10 -DPCBREV=T18 -DINTERNAL_MODULE_MULTI=YES -DDEFAULT_MODE=2 -DMODULE_PROTOCOL_FLEX=YES -DHARDWARE_TOUCH=YES -DUSB_CHARGER=YES -DLUA=YES ..

The expected end result after running make flash would be "welcome to opentx" after pressing power, but it stops before that happens and only shows the logo.

How do I properly build it for the specified target?

**Comment by sfjuocekr** [comment]
I figured out what was wrong with my builds, I'll finish off the changes needed and add TX18S build support in the following days (if there are no objections against it).

**Comment by kyon03** [comment]
nice

**Comment by sfjuocekr** [comment]
I never continued working on it, I've tried to contact RadioKing as well to see if they were willing to release the sourcecode. Sadly, they have not responded adequately :(

For now I can recommend just using the Jumper T18 build, the radio is basically a T18 with added LCD controller and some changes to the way audio is played back. At least, I do not get the hissing sound with stock RadioKing firmware but on the T18 firmware I do get it. The hissing stops temporarily if you enter the global menu, but it does not bother me.

**Comment by sfjuocekr** [comment]
The changes are up on my github page btw, if someone wants to mess about with it.

**Comment by RipperGOT** [comment]
> The changes are up on my github page btw, if someone wants to mess about with it.

Hi!  What are the changes you made for?  Another question, do you have jitter problems?  
Check out this link. 
[https://github.com/EdgeTX/edgetx/issues/702]

**Comment by ruedig** [comment]
When delivered, my Radioking TX18S showed the correct screen view for the boot loader and also for the model settings.
After the upload to OpenTX 2.3.14, the view of the bootloader screen is shown rotated by 180 °.

![image](https://user-images.githubusercontent.com/93605932/139908424-f5b3481d-c30b-4141-b7ec-1b0e1d8838ab.png)

How can I change this ??
Flashing back to OpenTX 2.3.10 was unsuccessful.

Thanks for the help
Rudiger

**Comment by sfjuocekr** [comment]
Use the T18 build, I can highly recommend you switch to EdgeTX.

**Comment by anshchawla521** [comment]
Hi @sfjuocekr  did u manage to get the touch screen working on RK TX18S ?

**Comment by sfjuocekr** [comment]
Yes, but I recommend you switch to EdgeTX and just flash the T18 build.
